File: apps/web/src/services/reading.ts
/**
 * 阅读理解 API
 */
import { config, fetchWithFallback } from '../config'

export interface ReadingChoice {
  label: string
  text: string
}

export interface ReadingQuestion {
  question: string
  choices: ReadingChoice[]
  correct_label: string
  explanation: string
}

export interface ReadingResult {
  summary: string
  questions: ReadingQuestion[]
  is_mock: boolean
}

export async function generateReading(
  passage: string,
  numQuestions: number = 3,
  level: string = 'medium',
): Promise<ReadingResult> {
  const proxyUrl = '/api/ai/reading'
  const directUrl = config.aiGateway
    ? `${config.aiGateway}/api/ai/reading`
    : proxyUrl

  const resp = await fetchWithFallback(proxyUrl, directUrl, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ passage, num_questions: numQuestions, level }),
  })

  if (!resp.ok) {
    const body = await resp.text().catch(() => '')
    throw new Error(`阅读理解生成失败 ${resp.status}: ${body.slice(0, 100)}`)
  }
  return resp.json()
}

/* ============ 官方真题库 ============ */

export interface BankItem {
  id: string
  exam: string
  source: string
  testNo: number | null
  module: number | null
  questionNo: number | null
  passage: string
  question: string
  choices: ReadingChoice[]
  correctLabel: string
  explanation: string
}

export interface BankResult {
  items: BankItem[]
  total: number
  exam: string
  source: string
}

export async function fetchReadingBank(
  params: { limit?: number; offset?: number; shuffle?: boolean } = {},
): Promise<BankResult> {
  const qs = new URLSearchParams()
  qs.set('limit', String(params.limit ?? 10))
  if (params.offset) qs.set('offset', String(params.offset))
  if (params.shuffle) qs.set('shuffle', 'true')

  const path = `/api/reading/bank?${qs.toString()}`
  const directUrl = config.aiGateway ? `${config.aiGateway}${path}` : path

  const resp = await fetchWithFallback(path, directUrl, { method: 'GET' })
  if (!resp.ok) throw new Error(`真题库加载失败 ${resp.status}`)
  return resp.json()
}

File: services/ai-gateway/app/prompts/reading.py
"""阅读理解 Prompt"""
from __future__ import annotations

READING_SYSTEM = """你是一位英语阅读理解出题专家。

任务：根据给定的英文文章，出 {num} 道阅读理解选择题，并写一段中文摘要。

要求：
1. 题目考察文章主旨、细节、推断、词汇含义等
2. 每题 4 个选项（1 正确 + 3 干扰），干扰项合理
3. 正确答案有简洁的中文解析

输出 JSON（仅 JSON）：
{{
  "summary": "文章中文摘要（50-100字）",
  "questions": [
    {{
      "question": "题目（英文）",
      "choices": [
        {{"label": "A", "text": "选项A"}},
        {{"label": "B", "text": "选项B"}},
        {{"label": "C", "text": "选项C"}},
        {{"label": "D", "text": "选项D"}}
      ],
      "correct_label": "A",
      "explanation": "中文解析"
    }}
  ]
}}
"""


def build_reading_messages(passage: str, num: int, level: str) -> list[dict]:
    system = READING_SYSTEM.format(num=num)
    user = f"难度：{level}\n\n文章：\n{passage}\n\n请出题。"
    return [
        {"role": "system", "content": system},
        {"role": "user", "content": user},
    ]